rfd/0164-scoped-rbac.md

```diff
@@ -0,0 +1,956 @@
+
+#### Roles and Access Lists in resource groups
+
+A special case is when a role or an access list is assigned to a certain resource group.  
+
+Only roles that have `grantable_scope` matching the resource group can be assigned to the resource group.
+
+The same applies to access lists, the scope of the access list grants should always match the scope of the roles it grants access to and 
+should not exceed the scope of the access list itself.
+
+In both of those cases, parent resource group must be specified both for access lists and roles and should equal or be more specific than the scope it was created in:
+
+```yaml
+kind: role
+metadata:
+  name: lab-admin
+spec:
+  grantable_scope: /env/prod/lab
+  parent_resource_group: /env/prod/lab
+```
+
+```yaml
+kind: access_list
+metadata:
+  name: lab-personnel
+spec:
+  grants_scope: /env/prod/lab
+  parent_resource_group: /env/prod/lab
+```
+
+Roles created within a scope will have `grantable_scope` and `parent_resource_group` to be equal to scope, or more specific. 
+
+For example, any role created within a scope `/env/prod/lab` must have the same `grantable_scope` and `parent_resource_group`  - `/env/prod/lab` or more specific one, 
+for example `/env/prod/lab/west-wing`.
+
+The roles and access lists created in the scope must grant access to scopes equal, or more specific than the scope they were created in to prevent lateral expansion or permissions.
+
+These invariants will let us make sure that any role created within a scope, will only grant permissions in the same, or more specific scope. 
+
+We will apply the same invariants to any other resources created within a scope. 
+
+#### Scoped Join Tokens
+
+Join tokens with a `parent_resource_group` will bind any resource using this join method to the resource group specified in `parent_resource_group` or a more specific one. 
+
+```yaml
+# token.yaml
+kind: token
+version: v2
+metadata:
+  name: my-token-name
+spec:
+  parent_resource_group: '/dev'
+  roles: 
+    - Node
+    - App
+```
+
+Join tokens created by roles granted within a scope must have `parent_resource_group` equal to this scope `/dev` or a more specific scope, e.g. `/dev/lab`
+
+#### Access Requests
+
+Access requests are bound to the scope they are created within, when Alice requests access to environment `/dev/lab` with role access, 
+the access request will capture the scope `/dev/lab` and create grant at this scope for Alice when approved.
+
+#### Scoped Audit Log Events and Session Recordings
+
+The audit events and session recordings generated by activity in some scope, will have a property that binds them to the same scope. 
+
+This will let us filter and grant access to a subset of events and session recordings within a scope.
+
+#### Resources that can’t be created at non-cluster scopes
+
+Some resources don't have a clear cut behavior at certain scopes, like SSO connectors, or are difficult to define, like users. 
```

Comment:
should we separate these global resources out of the role and we keep the roles for scoped resources.

Cluster-wide configurations for most resources should be tracked using a different resource type to prevent any incorrect concealment of permissions like the one described in https://github.com/gravitational/teleport.e/issues/3111

Reply:
Interesting idea, let me think about it.

Reply:
I've introduced an `admin:` scope to address this issue instead of creating a separate resource, take a look.